src/pal/src/include/pal/stackstring.hpp

```diff
@@ -44,45 +50,64 @@ class StackString
             return;
         }
 
+        if(preserve)
+        {
+            CopyMemory(newBuffer, m_buffer, (m_count + 1) * sizeof(T));
+        }
+
         DeleteBuffer();
         m_buffer = newBuffer;
         m_count = count;
-        m_size = count+1;
+        m_size = count_allocated+1;
+        isConstant = FALSE;
 
         return;
     }
+    
+    BOOL HasAvaliableMemory(SIZE_T count)
+    {
+        if(isConstant == TRUE)
+        {
+            return FALSE;
+        }
+
+        if (m_count + 1 != m_size)
```

Comment:
Why check `m_count + 1 != m_size` before `count < m_size`? Directly using `return (count < m_size)` should be fine.